pkg/datastore/bootstrap.go

```diff
@@ -4,36 +4,103 @@ import (
 	"context"
 	"fmt"
 	gyaml "github.com/ghodss/yaml"
+	"github.com/golang/protobuf/ptypes"
+	healthv1 "github.com/metal-stack/masterdata-api/api/grpc/health/v1"
+	v1 "github.com/metal-stack/masterdata-api/api/v1"
+	"github.com/metal-stack/masterdata-api/pkg/health"
 	"go.uber.org/zap"
 	"io/ioutil"
 	"path"
 	"path/filepath"
 	"reflect"
 	"regexp"
 	"strings"
-
-	healthv1 "github.com/metal-stack/masterdata-api/api/grpc/health/v1"
-	v1 "github.com/metal-stack/masterdata-api/api/v1"
-	"github.com/metal-stack/masterdata-api/pkg/health"
 )
 
 // Initdb reads all yaml files in given directory and apply their content as initial datasets.
-func (d *Datastore) Initdb(healthServer *health.Server, dir string) error {
-	files, err := d.listFiles(dir)
+func (ds *Datastore) Initdb(healthServer *health.Server, dir string) error {
+	files, err := ds.listFiles(dir)
 	if err != nil {
 		return err
 	}
 	for _, f := range files {
-		d.log.Info("read initdb", zap.Any("file", f))
-		err = d.processConfig(f)
+		ds.log.Info("read initdb", zap.Any("file", f))
+		err = ds.processConfig(f)
 		if err != nil {
 			return err
 		}
 	}
+
+	entities := []interface{}{&[]v1.Project{}, &[]v1.Tenant{}}
+	for _, e := range entities {
+		err = ds.consolidateHistory(e)
+		if err != nil {
+			ds.log.Error("error consolidate history", zap.Error(err))
+		}
+	}
+
 	healthServer.SetServingStatus("initdb", healthv1.HealthCheckResponse_SERVING)
 	return nil
 }
 
+// consolidateHistory ensures, that for each VersionedJSONEntity there is at least one "created"-row in the history table.
```

Comment:
As far as i understood you problem statement, you would like to create a history entry for all entries in the current table if none exists, but iterating over the configuration file will not cope these entries which have been created via api call before the history functionality was implemented.
Wouldnt it be better to iterate over all of the the current entries and check if a history entry for `created` exists and create one if not ?

Reply:
This is exactly what I do. The consolidation of the history table is done by checking the existing rows in the entity-tables for missing created-rows and creating them, there is no relation to the config-file processing, except that it is triggered in a func named initdb() and the comment above the func is now misleading.

Should I create another func "migrateDB" or something that makes it clear that the consolidation is something separate and introduce a separate servingStatus.

Reply:
OK, got it, was confused by the changes to the listFiles call.
This approach will of course be always run, and is only required once, should we introduce a version table which is incremented on every major table related change and run this part of the code if version is smaller than required ?

Reply:
I see your point, will try to implement a lean mechanism

Reply:
Please take a look